### PR TITLE
Remove Safari 14 support in Wagtail admin – update browserslist and TypeScript

### PR DESCRIPTION
**Pull request by thibaudcolas**
See #8520. This removes Safari 14 support code in the admin. Includes updating our `browserslist` / `caniuse-lite`, and TypeScript tranpiling targets.

For browserslist here is a diff of the changes in support targets based on my changes:

```diff
-and_chr 116
+and_chr 119
-chrome 116
-chrome 115
+chrome 118
+chrome 119
-edge 116
-edge 115
+edge 118
+edge 119
-firefox 116
+firefox 119
firefox 91
firefox 78
-ios_saf 16.5
-ios_saf 16.4
+ios_saf 17.1
+ios_saf 17.0
+ios_saf 16.0
-safari 16.5
-safari 16.4
-safari 16.3
+safari 17.1
+safari 17.0
+safari 16.6
+safari 15
```

Specifying Safari support targets as "last X" isn’t enough, as browserslist also applies this to minor releases. In our cases, our "last 3 Safari macOS versions" mean 17, 16, 15 – not 17.1, 17.0, 16.6.

I have also removed IE11 support notes, as they’re no longer up-to-date nor relevant.

_Please check the following:_

-   [ ] Do the tests still pass?[^1]
-   [ ] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

Tested with the [wagtail-tooling visual regression test suite _in Chrome_](https://github.com/thibaudcolas/wagtail-tooling), and with manual exploratory navigation of the admin.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,7 +100,7 @@
     "tailwindcss-vanilla-rtl": "^0.2.0",
     "ts-jest": "^29.1.0",
     "ts-loader": "^9.2.6",
```
Comment (laymonage) [inline package.json:102]: Looks like we need to update `ts-loader` as well to get the build working again:

```suggestion
    "ts-loader": "^9.5.1",
```

**Comment by laymonage** [comment]
Here's my branch with TypeScript 5 and Storybook v7: https://github.com/laymonage/wagtail/tree/storybook-7

**Comment by lb-** [comment]
Pumped about the TypeScript update. Happy to review this tomorrow if it's not already done by then.

**Review by laymonage — APPROVED** [review]
Generally happy with this, though we may need to change the browserslist definition for iOS as we discussed. Thanks!

**Comment by lb-** [comment]
Hey, I am getting the following linting warnings locally after this update. This just adds a bit of noise when running linting.


## A - typescript-estree

I think this relates to using the Storybook linting package, which uses an older version of estlint typescript utils `@typescript-eslint/utils`. Looks like there is already issue raised here to update https://github.com/storybookjs/eslint-plugin-storybook/issues/138

This is not a blocking issue so hopefully it will resolve itself in a future update of the storybook linting package.

```
> wagtail@1.0.0 lint:js
> eslint --ext .js,.ts,.tsx --report-unused-disable-directives .

=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=4.3.5 <5.2.0

YOUR TYPESCRIPT VERSION: 5.2.2

Please only submit bug reports when using the officially supported version.

=============
```

## B. TS checks failing due to inner dependency

This one is not easily solvable with an update, it appears that Popper.js has not been updated in a few years and has not kept up with changes to the popper/core library updates.

The fact that this did not fail the CI is confusing to me, I thought `tsc --noEmit` should fail as a linting step and have caught this earlier.

Anyway, for now, I think that whether or not this fails the CI, we should avoid TSC checking inner dependencies, it should be only checking and running against the direct usage of libraries (e.g. tippy and not popper/core in our case).

I have raised https://github.com/wagtail/wagtail/pull/11336 to avoid this issue.

```
> wagtail@1.0.0 lint:ts
> tsc --noEmit

node_modules/@popperjs/core/lib/types.d.ts:123:34 - error TS2344: Type 'Options' does not satisfy the constraint 'Obj'.

123     fn: (arg0: ModifierArguments<Options>) => State | void;
                                     ~~~~~~~

  node_modules/@popperjs/core/lib/types.d.ts:117:36
    117 export declare type Modifier<Name, Options> = {
                                           ~~~~~~~
    This type parameter might need an `extends Obj` constraint.

node_modules/@popperjs/core/lib/types.d.ts:124:39 - error TS2344: Type 'Options' does not satisfy the constraint 'Obj'.

124     effect?: (arg0: ModifierArguments<Options>) => (() => void) | void;
                                          ~~~~~~~

  node_modules/@popperjs/core/lib/types.d.ts:117:36
    117 export declare type Modifier<Name, Options> = {
                                           ~~~~~~~
    This type parameter might need an `extends Obj` constraint.


Found 2 errors in the same file, starting at: node_modules/@popperjs/core/lib/types.d.ts:123

```

**Comment by thibaudcolas** [comment]
@lb- I believe I’ve fixed the TypeScript version warning in #11354. It seems to work even with `eslint-plugin-storybook` using an older version.

If `eslint-plugin-storybook` is indeed causing issues, I’d be tempted for us to try npm `overrides` for a scenario like this.

**Comment by lb-** [comment]
Thanks Thibaud. I'll run the code when I get a chance and see if there are any other questions. Thanks for following this up

**Pull request merged**